pallets/proxy-financial/src/functions.rs

```diff
@@ -763,6 +1004,7 @@ impl<T: Config> Pallet<T> {
         // Ensure transactions are not empty
         ensure!(!transactions.is_empty(), Error::<T>::EmptyTransactions);
```

Comment:
This validation should be inside the transactions, and the `for transaction in transactions` would be only used if transactions are been made. Just to let the user sign the extrinsic when he wants to move the drawdown to the next phase.